### PR TITLE
add RCUTILS_LOGGING_AUTOINIT_WITH_ALLOCATOR.

### PR DESCRIPTION
**Pull request by fujitatomoya**
address https://github.com/ros2/rcl/issues/1036

### CODE REVIEW

File: include/rcutils/logging.h [hunk 1]
```diff
@@ -571,6 +571,26 @@ void rcutils_logging_console_output_handler(
     } \
   } while (0)
 
+/**
+ * \def RCUTILS_LOGGING_AUTOINIT_WITH_ALLOCATOR
+ * \brief Initialize the rcl logging library with allocator.
+ * Usually it is unnecessary to call the macro directly.
+ * All logging macros ensure that this has been called once.
+ */
+#define RCUTILS_LOGGING_AUTOINIT_WITH_ALLOCATOR(alloc)       \
```
Comment (fujitatomoya) [inline include/rcutils/logging.h:580]: This is the same counterpart between `rcutils_logging_initialize` and `rcutils_logging_initialize_with_allocator`.

**Review by iuhilnehc-ynos — APPROVED** [review]

File: include/rcutils/logging.h (hunk 1 above)
Comment (fujitatomoya) [reply include/rcutils/logging.h:580]: @clalancette that is good idea, thanks!

Comment (fujitatomoya) [reply include/rcutils/logging.h:569]: addressed with https://github.com/ros2/rcutils/commit/1385a23032c230525ec1cd65e57508adf1951e19

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18258/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12794/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18933/)

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18275/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12805/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18945/)

**Review by clalancette — APPROVED** [review]

**Comment by fujitatomoya** [comment]
https://ci.ros2.org/job/ci_windows/18945/#showFailuresLink unstable failure is not related to this PR.

**Pull request merged**